diff -Naur ocamldap-2.1.8/ldap_funclient.ml ocamldap-2.1.8/ldap_funclient.ml
--- ocamldap-2.1.8/ldap_funclient.ml	2006-06-16 02:36:39.000000000 +0100
+++ ocamldap-2.1.8/ldap_funclient.ml	2013-06-19 16:39:48.126149000 +0100
@@ -161,26 +161,26 @@
 		   if mech = `PLAIN then
 		     let s = socket PF_INET SOCK_STREAM 0 in
 		       try
-			 previous_signal :=
+			 (*previous_signal :=
 			   signal sigalrm 
 			     (Signal_handle (fun _ -> failwith "timeout"));
-			 ignore (alarm connect_timeout);
+			 ignore (alarm connect_timeout);*)
 			 connect s (ADDR_INET (addr, port));
-			 ignore (alarm 0);
-			 set_signal sigalrm !previous_signal;
+			 (*ignore (alarm 0);
+			 set_signal sigalrm !previous_signal;*)
 			 Plain s
 		       with exn -> close s;raise exn
 		   else
-		     (previous_signal := 
+		     ((*previous_signal := 
 			signal sigalrm 
 			  (Signal_handle (fun _ -> failwith "timeout"));
-		      ignore (alarm connect_timeout);
+		      ignore (alarm connect_timeout);*)
 		      let ssl = Ssl (Ssl.open_connection 
 				       Ssl.SSLv23 
 				       (ADDR_INET (addr, port))) 
 		      in
-			ignore (alarm 0);
-			set_signal sigalrm !previous_signal;
+			(*ignore (alarm 0);
+			set_signal sigalrm !previous_signal;*)
 			ssl)
 		 with
 		     Unix_error (ECONNREFUSED, _, _) 
diff -Naur ocamldap-2.1.8/ldap_protocol.ml ocamldap-2.1.8/ldap_protocol.ml
--- ocamldap-2.1.8/ldap_protocol.ml	2006-06-16 02:36:39.000000000 +0100
+++ ocamldap-2.1.8/ldap_protocol.ml	2013-06-19 16:39:48.126149000 +0100
@@ -311,7 +311,7 @@
 let encode_bindresponse {bind_result=result;bind_serverSaslCredentials=saslcred} =
   let encoded_result = encode_components_of_ldapresult result in
   let encoded_saslcred = (match saslcred with
-			      Some s -> Some (encode_ber_octetstring s)
+           Some s -> Some (encode_ber_octetstring ~cls:Context_specific ~tag:7 s)
 			    | None -> None)
   in
   let len = (String.length encoded_result) + 
diff -Naur ocamldap-2.1.8/Makefile ocamldap-2.1.8/Makefile
--- ocamldap-2.1.8/Makefile	2006-06-16 02:36:39.000000000 +0100
+++ ocamldap-2.1.8/Makefile	2013-06-19 16:40:18.407406000 +0100
@@ -12,7 +12,7 @@
 ldif_changerec_parser.mly ldif_changerec_lexer.mll			\
 ldif_changerec_oo.mli ldif_changerec_oo.ml ldap_toplevel.ml
 RESULT=ocamldap
-PACKS=netstring str ssl
+PACKS=netstring str ssl compiler-libs.toplevel
 #OCAMLFLAGS=-rectypes
 
 LIBINSTALL_FILES=$(wildcard *.mli *.cmi *.cma *.cmxa *.a *.so *.o *.cmx ldap_toplevel.cmo)
diff -Naur ocamldap-2.1.8/META ocamldap-2.1.8/META
--- ocamldap-2.1.8/META	2006-06-16 02:36:39.000000000 +0100
+++ ocamldap-2.1.8/META	2013-06-19 16:39:48.102711500 +0100
@@ -1,4 +1,4 @@
 requires="netstring str ssl"
-version="2.1.5"
+version="2.1.8"
 archive(byte)="ocamldap.cma"
 archive(native)="ocamldap.cmxa"
